File: frontend/src/types/trip.ts
export interface Trip {
  id: string
  code: string
  title: string
  subtitle?: string
  destination: string
  destinations_detail: string[]
  country: string
  province?: string
  city?: string
  category: string
  duration_days: number
  duration_nights?: number
  departure_city?: string
  best_season?: string
  price_adult?: number
  price_child?: number
  price_infant?: number
  single_room_supplement?: number
  price_includes: string[]
  price_excludes: string[]
  summary?: string
  highlights: string[]
  recommendation_reasons: string[]
  detailed_description?: string
  group_size_min: number
  group_size_max?: number
  departure_dates: string[]
  cover_image_url?: string
  image_urls: string[]
  brochure_url?: string
  is_featured: boolean
  status: string
  schedules: TripSchedule[]
  created_at: string
  updated_at: string
}

export interface TripSchedule {
  id: string
  trip_id: string
  day_number: number
  theme?: string
  schedule_type: string
  time_start?: string
  time_end?: string
  location?: string
  activity?: string
  description?: string
  meal_included?: string
  hotel_name?: string
  hotel_stars?: number
  hotel_description?: string
  transport_type?: string
  transport_detail?: string
  tips?: string
  image_urls: string[]
  sort_order: number
}

export interface Order {
  id: string
  order_code: string
  trip_id: string
  trip_title: string
  trip_destination: string
  departure_date?: string
  num_adults: number
  num_children: number
  num_infants: number
  total_price?: number
  paid_amount: number
  discount_amount: number
  participants?: any[]
  status: string
  payment_status: string
  contract_url?: string
  insurance_url?: string
  created_at: string
  updated_at: string
}

export interface ScriptSegment {
  timecode_start: string
  timecode_end: string
  duration_seconds: number
  segment_type: 'hook' | 'highlights' | 'detail' | 'cta'
  text: string
  image_ref?: string
  image_keyword?: string
  bgm_suggestion?: string
}

export interface VideoScript {
  id: string
  trip_id: string
  title?: string
  platform: string
  duration_seconds: number
  style?: string
  script_content: string
  segments: ScriptSegment[]
  script_json?: any
  hook_text?: string
  highlights_text?: string
  detail_text?: string
  cta_text?: string
  image_assignments?: any
  quality_score?: number
  engagement_score?: number
  accuracy_score?: number
  completeness_score?: number
  generation_version: number
  polish_iterations: number
  status: string
  capcut_draft_id?: string
  export_status: string
  exported_at?: string
  created_at: string
  updated_at?: string
}

export interface ScriptGenerateRequest {
  trip_id: string
  platform?: string
  duration_seconds?: number
  style?: string
}

export interface ScriptPolishRequest {
  focus_areas?: string[]
  target_segment_index?: number
}

export interface ScriptEvaluateResponse {
  quality_score: number
  engagement_score: number
  accuracy_score: number
  completeness_score: number
  passed: boolean
  feedback: string
}

export interface ScriptExportResponse {
  draft_id: string
  draft_name: string
  file_size_bytes: number
  platform_style: Record<string, unknown>
  instructions: string
}

export interface ScriptListResponse {
  id: string
  trip_id: string
  title?: string
  platform: string
  duration_seconds: number
  quality_score?: number
  status: string
  export_status: string
  created_at: string
}

export interface RecommendResult {
  trip_id: string
  title: string
  destination: string
  duration_days: number
  price_adult?: number
  highlights: string[]
  cover_image_url?: string
  match_score: number
  match_reasons: string[]
  match_tags: string[]
  category: string
}

File: backend/app/schemas/trip.py
"""Trip schemas for request/response validation."""

from datetime import date, time
from typing import Optional
from decimal import Decimal

from pydantic import BaseModel, Field


# ── Trip Schedule ──

class ScheduleBase(BaseModel):
    day_number: int = Field(..., ge=1, le=30)
    theme: Optional[str] = None
    schedule_type: str = Field(default="景点")
    time_start: Optional[time] = None
    time_end: Optional[time] = None
    location: Optional[str] = None
    activity: Optional[str] = None
    description: Optional[str] = None
    meal_included: Optional[str] = None
    hotel_name: Optional[str] = None
    hotel_stars: Optional[int] = None
    hotel_description: Optional[str] = None
    transport_type: Optional[str] = None
    transport_detail: Optional[str] = None
    tips: Optional[str] = None
    image_urls: list[str] = []
    sort_order: int = 0


class ScheduleCreate(ScheduleBase):
    pass


class ScheduleResponse(ScheduleBase):
    id: str
    trip_id: str

    class Config:
        from_attributes = True


# ── Trip ──

class TripBase(BaseModel):
    code: str = Field(..., max_length=50)
    title: str = Field(..., max_length=200)
    subtitle: Optional[str] = None
    destination: str = Field(..., max_length=100)
    destinations_detail: list[str] = []
    country: str = "中国"
    province: Optional[str] = None
    city: Optional[str] = None
    category: str = "国内游"
    duration_days: int = Field(..., ge=1)
    duration_nights: Optional[int] = None
    departure_city: Optional[str] = None
    best_season: Optional[str] = None
    price_adult: Optional[float] = None
    price_child: Optional[float] = None
    price_infant: Optional[float] = None
    single_room_supplement: Optional[float] = None
    price_includes: list[str] = []
    price_excludes: list[str] = []
    detailed_description: Optional[str] = None
    group_size_min: int = 1
    group_size_max: Optional[int] = None
    departure_dates: list[str] = []
    cover_image_url: Optional[str] = None
    image_urls: list[str] = []
    is_featured: bool = False


class TripCreate(TripBase):
    """Create a new trip."""
    schedules: list[ScheduleCreate] = []


class TripUpdate(BaseModel):
    """Update an existing trip (all fields optional)."""
    title: Optional[str] = None
    subtitle: Optional[str] = None
    destination: Optional[str] = None
    category: Optional[str] = None
    duration_days: Optional[int] = None
    price_adult: Optional[float] = None
    price_child: Optional[float] = None
    detailed_description: Optional[str] = None
    cover_image_url: Optional[str] = None
    image_urls: Optional[list[str]] = None
    departure_dates: Optional[list[str]] = None
    status: Optional[str] = None
    is_featured: Optional[bool] = None


class TripListResponse(BaseModel):
    """Trip list item."""
    id: str
    code: str
    title: str
    subtitle: Optional[str]
    destination: str
    category: str
    duration_days: int
    price_adult: Optional[float]
    cover_image_url: Optional[str]
    summary: Optional[str]
    highlights: list[str]
    is_featured: bool
    status: str
    created_at: str

    class Config:
        from_attributes = True


class TripDetailResponse(TripListResponse):
    """Full trip detail."""
    destinations_detail: list[str]
    country: str
    province: Optional[str]
    city: Optional[str]
    duration_nights: Optional[int]
    departure_city: Optional[str]
    best_season: Optional[str]
    price_child: Optional[float]
    price_infant: Optional[float]
    single_room_supplement: Optional[float]
    price_includes: list[str]
    price_excludes: list[str]
    recommendation_reasons: list[str]
    detailed_description: Optional[str]
    group_size_min: int
    group_size_max: Optional[int]
    departure_dates: list[str]
    image_urls: list[str]
    brochure_url: Optional[str]
    schedules: list[ScheduleResponse] = []
    updated_at: str

    class Config:
        from_attributes = True


class TripSummaryRequest(BaseModel):
    """Request AI summary generation for a trip."""
    regenerate: bool = False
